src/api/auth.ts

```diff
@@ -69,8 +74,15 @@ export function middleware(req: AuthRequest, res: express.Response,
 }
 
 export function hasAccess(req: AuthRequest, res: express.Response,
-  next: express.NextFunction): void {
-  next();
+  next: express.NextFunction): void | express.Response {
+    let auth = getAuth();
+    let user = auth.users
+      .find(u => u.tokens.findIndex(t => t === res.locals.remote_user.token) != -1);
```

Comment:
I mean with double equations, `!==` instead of `!=`, that means you are also checking type of the variable.